lib/uklibid/include/uk/libid.h

```diff
@@ -6,4 +6,53 @@
 #ifndef LIBUKLIBID_H_
 #define LIBUKLIBID_H_
 
+#include <uk/config.h>
+#include <uk/arch/limits.h>
+#include <uk/bits/libid.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define UKLIBID_NONE __UI_MAX
+
+#define _uk_libid_static(l) \
+	__UKLIBID_ ## l
+
+/**
+ * Compile-time mapping of a library name to its unique library identifier
+ * WARNING: Any unknown library name (e.g., library not part of build) will
+ *          make this pre-processor macro fail.
+ *
+ * @param libname Non-quoted name of the library (no C-string!)
+ * @return Library identifier (unsigned int)
+ */
+#define uk_libid_static(libname) \
+	_uk_libid_static(libname)
+
+#ifndef __LIBNAME__
```

Comment:
What is the use-case of this?

Reply:
Pre-processor: This allows `libname` to be resolved when an alias (via macro or macro function) is handed over as argument